src/server/game/AI/ScriptedAI/ScriptedCreature.cpp

```diff
@@ -547,6 +547,23 @@ void BossAI::UpdateAI(uint32 diff)
     DoMeleeAttackIfReady();
 }
 
+void BossAI::_DespawnAtEvade(uint32 delayToRespawn)
+{
+    uint32 corpseDelay = me->GetCorpseDelay();
+    uint32 respawnDelay = me->GetRespawnDelay();
+
+    me->SetCorpseDelay(1);
+    me->SetRespawnDelay(std::max(delayToRespawn, (uint32)2) - 1);
```

Comment:
Ew, C-style cast. `2u` does what you want, though a simple
```c++
if (delayToRespawn < 2)
{
    TC_LOG_ERROR("scripts", _DespawnAtEvade called with delay of %d seconds, defaulting to 2.", delayToRespawn);
    delayToRespawn = 2;
}
```
is cleaner - not as compact, of course, but we aren't playing code golf.